pages/_app.tsx

```diff
@@ -62,6 +64,18 @@ type ComponentWithPageLayout = AppProps & {
 };
 
 function MyApp({ Component, pageProps }: ComponentWithPageLayout) {
+  const faroRef = useRef<null | Faro>(null);
+
+  useEffect(() => {
+    if (
+      process.env.NEXT_PUBLIC_FARO_COLLECTOR_URL &&
+      process.env.NEXT_PUBLIC_FARO_APP_ENVIRONMENT != "local" &&
+      !faroRef.current
+    ) {
+      faroRef.current = initGrafanaFaro();
+    }
+  }, []);
```

Comment:
The useRef seems unnecessary. If all you're trying to do is initialize this for the client-side only, this should suffice.

```suggestion
  useEffect(() => {
    if (
      process.env.NEXT_PUBLIC_FARO_COLLECTOR_URL &&
      process.env.NEXT_PUBLIC_FARO_APP_ENVIRONMENT != "local"
    ) {
      initGrafanaFaro();
    }
  }, []);
```

Reply:
Yeah, that's what I tried initially but would see this pop up in:

```
Faro
 Faro is already registered. Either add instrumentations, transports etc. to the global faro instance or use the "isolate" property
```

But that must be due to a re-render that happens? Either way, seems to be fine since the points still get sent to the collector endpoint ok:

![Screenshot 2024-04-05 at 3 52 47 PM](https://github.com/open-sauced/app/assets/23109390/c20f3709-7859-4749-86ec-dd64528501d3)